app/src/main/java/com/seafile/seadroid2/cameraupload/CameraSyncAdapter.java

```diff
@@ -501,69 +534,120 @@ private String varArgs(int count) {
+            file_iter += 1;
+//            Utils.utilsLogInfo(true,"=======Processing File ["+ Integer.toString(file_iter) + "/" + Integer.toString(file_num) + "]=======");
+            Log.d(DEBUG_TAG,"=======Processing File ["+ Integer.toString(file_iter) + "/" + Integer.toString(file_num) + "]=======");
             if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
                 if (media.equals("images")) {
-                    String image_id = cursor.getString(cursor.getColumnIndexOrThrow(MediaStore.Images.Media._ID));
+                    int image_column_index = cursor.getColumnIndex(MediaStore.Images.Media._ID);
+                    file_name = cursor.getString(cursor.getColumnIndex(MediaStore.Images.Media.DISPLAY_NAME));
+                    file_modified = cursor.getInt(cursor.getColumnIndex(MediaStore.Images.Media.DATE_MODIFIED));
+                    file_size = cursor.getInt(cursor.getColumnIndex(MediaStore.Images.Media.SIZE));
+                    String image_id = cursor.getString(image_column_index);
                     Uri image_uri = Uri.withAppendedPath(MediaStore.Images.Media.EXTERNAL_CONTENT_URI, image_id);
+                    file_path = Utils.getRealPathFromURI(SeadroidApplication.getAppContext(), image_uri, media);
                     if (image_uri == null) {
                         syncResult.stats.numSkippedEntries++;
                         continue;
                     }
-                    file = new File(Utils.getRealPathFromURI(SeadroidApplication.getAppContext(), image_uri, media));
+//                    file = new File(Utils.getRealPathFromURI(SeadroidApplication.getAppContext(), image_uri, media));
                 } else {
-                    String video_id = cursor.getString(cursor.getColumnIndexOrThrow(MediaStore.Video.Media._ID));
+                    int video_column_index = cursor.getColumnIndex(MediaStore.Video.Media._ID);
+                    file_name = cursor.getString(cursor.getColumnIndex(MediaStore.Video.Media.DISPLAY_NAME));
+                    file_modified = cursor.getInt(cursor.getColumnIndex(MediaStore.Video.Media.DATE_MODIFIED));
+                    file_size = cursor.getInt(cursor.getColumnIndex(MediaStore.Video.Media.SIZE));
+                    String video_id = cursor.getString(video_column_index);
                     Uri video_uri = Uri.withAppendedPath(MediaStore.Video.Media.EXTERNAL_CONTENT_URI, video_id);
+                    file_path = Utils.getRealPathFromURI(SeadroidApplication.getAppContext(), video_uri, media);
                     if (video_uri == null) {
                         syncResult.stats.numSkippedEntries++;
                         continue;
                     }
-                    file = new File(Utils.getRealPathFromURI(SeadroidApplication.getAppContext(), video_uri, media));
+//                    file = new File(Utils.getRealPathFromURI(SeadroidApplication.getAppContext(), video_uri, media));
```

Comment:
Any reason to keep this line?

Reply:
(multiple times in this commit.)

Reply:
Keep this line just to help others change the code as they want.